components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/FirefoxAccount.kt

```diff
@@ -224,6 +224,125 @@ class FirefoxAccount(handle: FxaHandle, persistCallback: PersistCallback?) : Aut
+     *
+     * @param displayName The current device display name
+     */
+    fun setDeviceDisplayName(displayName: String) {
+        rustCall { e ->
+            LibFxAFFI.INSTANCE.fxa_set_display_name(this.handle.get(), displayName, e)
+        }
+    }
+
+    /**
+     * Retrieves the list of the connected devices in the current account, including the current one.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     */
+    fun getDevices(): Array<Device> {
+        val devicesBuffer = rustCall { e ->
+            LibFxAFFI.INSTANCE.fxa_get_devices(this.handle.get(), e)
+        }
+        try {
+            val devices = MsgTypes.Devices.parseFrom(devicesBuffer.asCodedInputStream()!!)
+            return Device.fromCollectionMessage(devices)
+        } finally {
+            LibFxAFFI.INSTANCE.fxa_bytebuffer_free(devicesBuffer)
+        }
+    }
+
+    /**
+     * Retrieves any pending commands for the current device.
+     * This should be called semi-regularly as the main method of commands delivery (push)
+     * can sometimes be unreliable on mobile devices.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     *
+     * @return A collection of [AccountEvent] that should be handled by the caller.
+     */
+    fun pollDeviceCommands(): Array<AccountEvent> {
+        val eventsBuffer = rustCall { e ->
+            LibFxAFFI.INSTANCE.fxa_poll_device_commands(this.handle.get(), e)
+        }
+        this.tryPersistState()
+        try {
+            val events = MsgTypes.AccountEvents.parseFrom(eventsBuffer.asCodedInputStream()!!)
+            return AccountEvent.fromCollectionMessage(events)
+        } finally {
+            LibFxAFFI.INSTANCE.fxa_bytebuffer_free(eventsBuffer)
+        }
+    }
+
+    /**
+     * Handle any incoming push message payload coming from the Firefox Accounts
+     * servers that has been decrypted and authenticated by the Push crate.
+     *
+     * This performs network requests, and should not be used on the main thread.
+     *
+     * @return A collection of [AccountEvent] that should be handled by the caller.
+     */
+    fun handlePushMessage(payload: String): Array<AccountEvent> {
+        val eventsBuffer = rustCall { e ->
+            LibFxAFFI.INSTANCE.fxa_handle_push_message(this.handle.get(), payload, e)
+        }
+        this.tryPersistState()
+        try {
+            val events = MsgTypes.AccountEvents.parseFrom(eventsBuffer.asCodedInputStream()!!)
+            return AccountEvent.fromCollectionMessage(events)
+        } finally {
+            LibFxAFFI.INSTANCE.fxa_bytebuffer_free(eventsBuffer)
+        }
+    }
+
+    /**
+     * Ensure the current device "Send Tab" commands has been registered with the server.
+     * This method should be called once per "device lifetime" after the Sync Keys have been
+     * obtained and called again if they change.
```

Comment:
Is "device lifetime" a sufficiently-well-defined term that we can use it here, or should we try to be more specific?  I wonder if we can explicitly enumerate the set of circumstances under which this should be called, e.g. "after successful signin", "after re-authenticating", maybe others as well?

Alternately, I believe that we can tell whether or not the send-tab command was registered from our internal state, would it be possible for us to take care of this re-registration automagically after device lifetime events?